posts/navarroaxel/21.git-rebase-force.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Recreating Commits with git rebase for Seamless Workflow
+published: true
+description: 'We dive into the advanced use of git rebase to recreate commits, especially useful when GitHub struggles with commit detection.'
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-rebase-force.png
+id: 2017205
+---
+
+In this series we've learned how to merge or split commits using the `git rebase` command, now it's time to learn how to recreate commits and when this can be useful. 🪄
+
+## Forcing the rebase
+
+By default, the rebase makes a _fast-forward_ when it's possible 🤔. This means that the commits aren't re-created when they're being applied over the same previous commit, keeping the commit ID (SHA) unchanged. You can force this recreation and get a new commit ID using the [`--force-rebase`](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---force-rebase) flag.
+
+```bash
+git rebase --force-rebase main
+```
+
+This may be useful in those sad days when GitHub has issues detecting new commits in the repository and fails to trigger the GitHub Actions or the pull request is out of sync with the Git repo. 🌚
+
+Also, this could be useful when you revert a failed merge and you need to remerge the same code now expecting to work properly 🤞 without _revert the reversion_  😵‍💫. Let's explain this complex scenario a little more.
```

Comment:
```suggestion
Also, this could be useful when you revert a failed merge and you need to remerge the same code now expecting to work properly 🤞 without _reverting the reversion_  😵‍💫. Let's explain this complex scenario a little more.
```